### ISSUE TITLE
Brainstorm way to avoid needing to check for css classes to inspect the state of the tree

### ISSUE THREAD

**Issue by jaltekruse** [mathquill/mathquill#971]
https://github.com/mathquill/mathquill/pull/947/files#r811339524

Comment from @mikehaverstock copied below from a PR to add some special case behavior in subscripts. Originally checked into the desmos fork here - https://github.com/desmosinc/mathquill/pull/129

Currently slated for inclusion in mainline through this PR - https://github.com/mathquill/mathquill/pull/947

This issue isn't to say this particular code shouldn't be in mathquill right now, but to document that we have some refactoring work to do if we want to avoid proliferating hacky fixes like this in other areas of the code.

```
    // Mathquill is gross. There are many different paths that
    // create subscripts and sometimes we don't even construct
    // true instances of `LatexCmds._`. Another problem is that
    // the relationship between the sub and the SupSub isn't
    // completely setup during a paste at the time we check
    // this. I wanted to use: `this.parent.parent.sub !== this.parent`
    // but that check doesn't always work. This seems to be the only
    // check that always works. I'd rather live with this than try
    // to change the init order of things.
```

**Comment by jaltekruse** [comment]
Another example as pointed out by @anandthakker - https://github.com/desmosinc/mathquill/pull/157#discussion_r675795036

return this.jQ[0].className === 'mq-binary-operator' ? 'plus' : 'positive';

I'm a little wary about using this.jQ[0].className to answer questions about how to interpret a node. I wonder if we should extract the logic for determining whether the operator is unary/binary from the determineOpClassType function just above here into a method on PlusMinus, and then use that instead of the className. cc @jwmerrill for a second opinion on this

This ended up being resolved to find a different method of inspecting the tree, but maybe we want to make the css classes harder to access after construction if this is a pattern we want to avoid? Maybe using private members now that we have typescript?
https://github.com/desmosinc/mathquill/pull/157/files#diff-45a476afc37a51b9e65473aaca2ca762110f5afc10b659c5cce8e9e4f7484ee4R792

**Comment by jwmerrill** [comment]
I agree that resorting to reading `className` here is not a good practice.

I'm a little skeptical about doing anything in the near term to make it actually impossible, though. The MQNodes hold onto and modify DOM, and I don't think that's going to be easy to hide without more big refactors.

I would be in favor of designing ways to answer this question (and similar questions) that never read DOM.
